recipes/open62541/all/conanfile.py

```diff
@@ -270,122 +277,141 @@ def _get_multithreading_option(self):
+
+        tc.variables["UA_ENABLE_JSON_ENCODING"] = self.options.json_support
+        tc.variables["UA_ENABLE_PUBSUB"] = self.options.pub_sub != False
+        tc.variables["UA_ENABLE_PUBSUB_ENCRYPTION"] = self.options.pub_sub_encryption != False
+
         if self.options.pub_sub != False:
             if self.settings.os == "Linux" and self.options.pub_sub == "Ethernet":
-                self._cmake.definitions["UA_ENABLE_PUBSUB_ETH_UADP"] = True
+                tc.variables["UA_ENABLE_PUBSUB_ETH_UADP"] = True
             elif self.settings.os == "Linux" and self.options.pub_sub == "Ethernet_XDP":
-                self._cmake.definitions["UA_ENABLE_PUBSUB_ETH_UADP_XDP"] = True
-        self._cmake.definitions["UA_ENABLE_DA"] = self.options.data_access
+                tc.variables["UA_ENABLE_PUBSUB_ETH_UADP_XDP"] = True
+        tc.variables["UA_ENABLE_DA"] = self.options.data_access
+
         if self.options.compiled_nodeset_descriptions == True:
-            self._cmake.definitions["UA_ENABLE_NODESET_COMPILER_DESCRIPTIONS"] = self.options.compiled_nodeset_descriptions
-            self._cmake.definitions["UA_NAMESPACE_ZERO"] = "FULL"
+            tc.variables["UA_ENABLE_NODESET_COMPILER_DESCRIPTIONS"] = self.options.compiled_nodeset_descriptions
+            tc.variables["UA_NAMESPACE_ZERO"] = "FULL"
         else:
-            self._cmake.definitions["UA_NAMESPACE_ZERO"] = self.options.namespace_zero
-        if tools.Version(self.version) < "1.3.1":
-            self._cmake.definitions["UA_ENABLE_MICRO_EMB_DEV_PROFILE"] = self.options.embedded_profile
-        self._cmake.definitions["UA_ENABLE_TYPENAMES"] = self.options.typenames
-        self._cmake.definitions["UA_ENABLE_STATUSCODE_DESCRIPTIONS"] = self.options.readable_statuscodes
-        self._cmake.definitions["UA_ENABLE_HARDENING"] = self.options.hardening
+            tc.variables["UA_NAMESPACE_ZERO"] = self.options.namespace_zero
+        if Version(self.version) < "1.3.1":
+            tc.variables["UA_ENABLE_MICRO_EMB_DEV_PROFILE"] = self.options.embedded_profile
+
+        tc.variables["UA_ENABLE_TYPENAMES"] = self.options.typenames
+        tc.variables["UA_ENABLE_STATUSCODE_DESCRIPTIONS"] = self.options.readable_statuscodes
+        tc.variables["UA_ENABLE_HARDENING"] = self.options.hardening
+
         if self.settings.compiler == "Visual Studio" and self.options.shared == True:
-            self._cmake.definitions["UA_MSVC_FORCE_STATIC_CRT"] = True
-        self._cmake.definitions["UA_COMPILE_AS_CXX"] = self.options.cpp_compatible
+            tc.variables["UA_MSVC_FORCE_STATIC_CRT"] = True
+
+        tc.variables["UA_COMPILE_AS_CXX"] = self.options.cpp_compatible
 
-        self._cmake.configure(source_dir=self._source_subfolder)
-        return self._cmake
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
```

Comment:
```suggestion
```

Reply:
resolved in af654d59664638865c7836e7c622c49f5f6b39f3